include/event_manager.h

```diff
@@ -37,6 +37,26 @@ extern "C" {
  *            false otherwise.
  */
 typedef bool (*cb_fn)(const struct event_header *eh);
+/**
+ * @brief List of bits in event type flags.
+ */
+enum event_type_flags {
+	/* Used to assert the user trying to set flags that should not be changed by the user */
```

Comment:
```
	EVENT_TYPE_FLAGS_SYSTEM_START, /* Flags set internally by event manager. */
	EVENT_TYPE_FLAGS_HAS_DYNDATA = EVENT_TYPE_FLAGS_SYSTEM_START,

	EVENT_TYPE_FLAGS_USER_SETTABLE_START, /* Flags available for user. */
	EVENT_TYPE_FLAGS_INIT_LOG_ENABLE = EVENT_TYPE_FLAGS_USER_SETTABLE_START,
	EVENT_TYPE_FLAGS_TRACEABLE,

	EVENT_TYPE_FLAGS_COUNT, /* Number of predefined flags. */

	EVENT_TYPE_FLAGS_USER_DEFINED_START = EVENT_TYPE_FLAGS_COUNT, /* Value greater or equal are available for user. */
```

or

```
	/* Flags set internally by event manager. */
	EVENT_TYPE_FLAGS_SYSTEM_START, 
	EVENT_TYPE_FLAGS_HAS_DYNDATA = EVENT_TYPE_FLAGS_SYSTEM_START,

	/* Flags available for user. */
	EVENT_TYPE_FLAGS_USER_SETTABLE_START,
	EVENT_TYPE_FLAGS_INIT_LOG_ENABLE = EVENT_TYPE_FLAGS_USER_SETTABLE_START,
	EVENT_TYPE_FLAGS_TRACEABLE,

	/* Number of predefined flags. */
	EVENT_TYPE_FLAGS_COUNT,

	/* Value greater or equal are available for user. */
	EVENT_TYPE_FLAGS_USER_DEFINED_START = EVENT_TYPE_FLAGS_COUNT,
```

Reply:
remove the comment from line 44, add comments close to each group of flags,
add group that can be used by the user (if we can) - these flags can be useful in hooks